src/main/java/uk/gov/hmcts/reform/ccd/client/healthcheck/InternalHealth.java

```diff
@@ -11,9 +10,9 @@
 
     @JsonCreator
     public InternalHealth(
-        @JsonProperty("status") Status status
```

Comment:
This didn't work, not sure why. Lost an hour on it and gave up.

Reply:
There are 2 gotcha's with this class:
- It has `@JsonProperty("status")` in there already
- Does not have default no arg constructor

Here's the snippet of working (I think) type safe code:

```java
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonRootName("status")
public class InternalHealth {
    @JsonUnwrapped
    private final Status status;

    @JsonCreator
    public InternalHealth(Status status) {
        this.status = status;
    }

    public Status getStatus() {
        return status;
    }
}

public class InternalHealthTest {

    private final ObjectMapper objectMapper = new ObjectMapper();

    @Before
    public void setUp() {
        objectMapper.enable(DeserializationFeature.UNWRAP_ROOT_VALUE);
    }

    @Test
    public void asd() throws IOException {
        InternalHealth health = objectMapper.readValue("{\"status\":\"UP\"}", InternalHealth.class);

        assertThat(health.getStatus()).isEqualTo(Status.UP);
    }

    @Test
    public void dsa() throws JsonProcessingException {
        String health = objectMapper.writeValueAsString(new InternalHealth(Status.DOWN));

        assertThat(health).isEqualTo("{\"status\":\"DOWN\"}");
    }
}
```